urllib3/response.py

```diff
@@ -182,6 +192,38 @@ def tell(self):
         """
         return self._fp_bytes_read
 
+    def _init_length(self, request_method):
+        """
+        Set initial length value for Response content.
+
+        Logic extracted from httplib.py in the standard Python library.
+        """
+        length = self.headers.get('content-length')
+        if length is not None and not self.chunked:
+            try:
+                length = int(length)
```

Comment:
This is not quite good enough, as it leads us to subtly disregard [RFC 7230 section 3.3.2](https://tools.ietf.org/html/rfc7230#section-3.3.2):

> If a message is received that has multiple Content-Length header fields with field-values consisting of the same decimal value, or a single Content-Length header field with a field value containing a list of identical decimal values (e.g., "Content-Length: 42, 42"), indicating that duplicate Content-Length header fields have been generated or combined by an upstream message processor, then the recipient MUST either reject the message as invalid or replace the duplicated field-values with a single valid Content-Length field containing that decimal value prior to determining the message body length or forwarding the message. 

We should aim to handle this case *either* by rejecting all invalid content lengths, *or* by handling this case specially. Doing neither is pretty bad though. =(